### PR TITLE
Add extractors for if, basic for and while statements

### PR DESCRIPTION
**Pull request by bryanlohxz**

### CODE REVIEW

File: src/ast/astExtractor/expression-extractor.ts [hunk 1]
```diff
@@ -312,6 +387,32 @@ export class ExpressionExtractor extends BaseJavaCstVisitorWithDefaults {
     return literal;
   }
 
+  booleanLiteral(ctx: BooleanLiteralCtx) {
+    return {
+      kind: "Literal",
+      literalType: {
+        kind: "BooleanLiteral",
+        value: ctx.False ? "false" : ("true" as "true" | "false"),
```
Comment (xyliew25) [inline src/ast/astExtractor/expression-extractor.ts:395]:
```suggestion
        value: ctx.False ? "false" : "true",
```

Comment (xyliew25) [reply src/ast/astExtractor/expression-extractor.ts:395]: Does this not work as well?

File: src/ast/types/blocks-and-statements.ts [hunk 2]
```diff
@@ -47,7 +57,7 @@ export type ForStatement = BasicForStatement | EnhancedForStatement;
 export interface BasicForStatement extends BaseNode {
   kind: "BasicForStatement";
   forInit: Array<ExpressionStatement> | LocalVariableDeclarationStatement;
-  condition: Expression;
+  condition: Expression | null;
```
Comment (xyliew25) [inline src/ast/types/blocks-and-statements.ts:60]:
```suggestion
  condition?: Expression;
```

Comment (xyliew25) [reply src/ast/types/blocks-and-statements.ts:60]: Do you meant to have `condition` as an optional field instead?

File: src/ast/astExtractor/block-statement-extractor.ts [hunk 3]
```diff
@@ -1,38 +1,42 @@
+import { ExpressionExtractor } from "./expression-extractor";
+import { Identifier, UnannType } from "../types/classes";
+import { StatementExtractor } from "./statement-extractor";
+import { TypeExtractor } from "./type-extractor";
```
Comment (xyliew25) [inline src/ast/astExtractor/block-statement-extractor.ts:4]: Why rearranged these imports up here? I thought they were alr in the correct positions?

File: src/ast/astExtractor/expression-extractor.ts [hunk 4]
```diff
@@ -162,29 +182,54 @@ export class ExpressionExtractor extends BaseJavaCstVisitorWithDefaults {
...
+        operator: suffixOp.image,
+        expression: node,
+        location: {
+          startOffset: suffixOp.startOffset,
+          startLine: suffixOp.startLine,
+          startColumn: suffixOp.startColumn,
+          endOffset: suffixOp.endOffset,
+          endLine: suffixOp.endLine,
+          endColumn: suffixOp.endColumn,
+        },
```
[the remark this answers is not in the document]
Comment (bryanlohxz) [reply src/ast/astExtractor/expression-extractor.ts:198]: No. Some nodes from `java-parser` like `Suffix` and `Image` don't have a `location` property. Instead, the location data is kept within the node itself like what is done here.

File: src/ast/astExtractor/expression-extractor.ts (hunk 1 above)
Comment (bryanlohxz) [reply src/ast/astExtractor/expression-extractor.ts:395]: In the case of your suggestion, the typing of `value` does not work as your suggestion will result in `value: string` while the above is `value: "true" | "false"`. However, I wonder why the assertion only applies to the alternative expression hmm...

**Review by 1001mei — APPROVED** [review]

**Pull request merged**